tests/attr/test_lrp.py

```diff
@@ -0,0 +1,314 @@
+        model.eval()
+        model.linear.rule = GammaRule()
+        model.linear2.rule = Alpha1_Beta0_Rule()
+        output = model(inputs)
+        lrp = LRP(model)
+        _ = lrp.attribute(inputs)
+        output_after = model(inputs)
+        assertTensorAlmostEqual(self, output, output_after)
+
+    def test_lrp_simple_inplaceReLU(self):
+        model_default, inputs = _get_simple_model()
+        model_inplace, _ = _get_simple_model(inplace=True)
+        for model in [model_default, model_inplace]:
+            model.eval()
+            model.linear.rule = EpsilonRule()
+            model.linear2.rule = EpsilonRule()
+        lrp_default = LRP(model_default)
+        lrp_inplace = LRP(model_inplace)
+        relevance_default = lrp_default.attribute(inputs)
+        relevance_inplace = lrp_inplace.attribute(inputs)
+        assertTensorAlmostEqual(self, relevance_default, relevance_inplace)
+
+    def test_lrp_simple_tanh(self):
+        class Model(nn.Module):
+            def __init__(self):
+                super(Model, self).__init__()
+                self.linear = nn.Linear(3, 3, bias=False)
+                self.linear.weight.data.fill_(0.1)
+                self.tanh = torch.nn.Tanh()
+                self.linear2 = nn.Linear(3, 1, bias=False)
+                self.linear2.weight.data.fill_(0.1)
+
+            def forward(self, x):
+                return self.linear2(self.tanh(self.linear(x)))
+
+        model = Model()
+        _, inputs = _get_simple_model()
+        _ = model(inputs)
+        lrp = LRP(model)
+        relevance = lrp.attribute(inputs)
+        # assertTensorAlmostEqual(
+        #   self, relevance, torch.Tensor([[0.1186, 0.2372, 0.3558]])
+        # )
+        # Result if gradient is used for propagation over tanh
+        assertTensorAlmostEqual(
+            self, relevance, torch.Tensor([[1 / 6, 1 / 3, 1 / 2]])
+        )  # Result if tanh is skipped for propagation
+
+    def test_lrp_simple_attributions_GammaRule(self):
+        model, inputs = _get_simple_model()
+        with torch.no_grad():
+            model.linear.weight.data[0][0] = -2
+        model.eval()
+        model.linear.rule = GammaRule(gamma=1)
+        model.linear2.rule = GammaRule()
+        lrp = LRP(model)
+        relevance = lrp.attribute(inputs)
+        assertTensorAlmostEqual(
+            self, relevance.data, torch.tensor([21 / 216, 78 / 216, 117 / 216])
+        )
+
+    def test_lrp_simple_attributions_AlphaBeta(self):
+        model, inputs = _get_simple_model()
+        with torch.no_grad():
+            model.linear.weight.data[0][0] = -2
+        model.eval()
+        model.linear.rule = Alpha1_Beta0_Rule()
+        model.linear2.rule = Alpha1_Beta0_Rule()
+        lrp = LRP(model)
+        relevance = lrp.attribute(inputs)
+        assertTensorAlmostEqual(self, relevance, torch.tensor([0.1250, 0.3500, 0.5250]))
+
+    def test_lrp_Identity(self):
+        model, inputs = _get_simple_model()
```

Comment:
Could we, please, add another test case for testing the point mentioned by @rGure here: https://github.com/pytorch/captum/pull/342/files#diff-ac7c4e84c72e959484504ba2f5467e49R141